src/index.js

```diff
@@ -154,3 +145,40 @@ function getDocument(rootNode) {
         return rootNode;
     }
 }
+
+function upgradeElement (element, definition, callConstructor) {
+    const prototype = definition.constructor.prototype;
+    Object.setPrototypeOf(element, prototype);
+    if (callConstructor) {
+        CustomElementRegistry.instance()._setNewInstance(element);
+        new (definition.constructor)();
+        element[_upgradedProp] = true;
+        console.assert(CustomElementRegistry.instance()._newInstance === null);
```

Comment:
Why do we have this assert in-line? Should probably either be removed or become a test.